### PR TITLE
[BugFix]Modify array_agg with sub error messages

### LINKED ISSUE #19198: unknown error when array_agg with sub
### Steps to reproduce the behavior (Required)
```
CREATE TABLE `t2` (
  `v1` varchar(20) NOT NULL COMMENT "",
  `v2` varchar(20) NOT NULL COMMENT "",
  `v3` varchar(20) NOT NULL COMMENT ""
) ENGINE=OLAP 
DUPLICATE KEY(`v1`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`v1`) BUCKETS 1 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"replicated_storage" = "false",
"compression" = "LZ4"
);

explain select array_agg(v1) - array_agg(v1) from t2;
```

### Expected behavior (Required)
shouldn't report unknown error

### Real behavior (Required)
```
unknown error
```
```
2023-03-08 14:54:57,682 WARN (starrocks-mysql-nio-pool-26|2425) [StmtExecutor.execute():544] execute Exception, sql explain select array_agg(v1) - array_agg(v1) from t2
java.lang.NullPointerException: null
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitArithmeticExpr(ExpressionAnalyzer.java:571) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitArithmeticExpr(ExpressionAnalyzer.java:224) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.ArithmeticExpr.accept(ArithmeticExpr.java:523) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:55) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.bottomUpAnalyze(ExpressionAnalyzer.java:221) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyzeExpression(ExpressionAnalyzer.java:1170) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.SelectAnalyzer.analyzeExpression(SelectAnalyzer.java:681) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.SelectAnalyzer.analyzeSelect(SelectAnalyzer.java:252) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.SelectAnalyzer.analyze(SelectAnalyzer.java:75) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:193) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:101) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.SelectRelation.accept(SelectRelation.java:236) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:106) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryRelation(QueryAnalyzer.java:124) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:118) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:101) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:49) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:106) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.analyze(QueryAnalyzer.java:94) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:347) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:137) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:49) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:55) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:139) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:134) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:65) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:50) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:389) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:349) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:466) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:730) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_202]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
```

### StarRocks version (Required)
 - UNKNOWN-270e30aa9d

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19198

## Problem Summary(Required) ：

fix after:
image

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by stephen-shelby — DISMISSED** [review]

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Review by Seaven — CHANGES_REQUESTED** [review]
I think we should to find why array can be cast to double/bigint, not throw expcetion direct

**Comment by leoyy0316** [comment]
> I think we should to find why array can be cast to double/bigint, not throw expcetion direct

ok, will check it

**Comment by leoyy0316** [comment]
> I think we should to find why array can be cast to double/bigint, not throw expcetion direct

@Seaven  i check it , not cast to double/int, in [visitArithmeticExpr] when getCommonType return Type.INVALID

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|[]|

**Comment by Seaven** [comment]
> > I think we should to find why array can be cast to double/bigint, not throw expcetion direct
> 
> @Seaven i check it , not cast to double/int, in [visitArithmeticExpr] when getCommonType return Type.INVALID

if common type is Type.INVALID, it's should check failed in line 626,  any type can't cast to Type.INVALID,  I think should to resolve this...

**Comment by Seaven** [comment]
> > > > I think we should to find why array can be cast to double/bigint, not throw expcetion direct
> > > 
> > > 
> > > @Seaven i check it , not cast to double/int, in [visitArithmeticExpr] when getCommonType return Type.INVALID
> > 
> > 
> > if common type is Type.INVALID, it's should check failed in line 626, any type can't cast to Type.INVALID, I think should to resolve this question...
> 
> @Seaven ok, i will modify. Also i think fn need to check failed when is null, maybe cause NPE

to check function is the correct way, but function is null in here is an unexpected error, I suggest add `Precondition.check` directly

**Comment by leoyy0316** [comment]
> > > > > I think we should to find why array can be cast to double/bigint, not throw expcetion direct
> > > > 
> > > > 
> > > > @Seaven i check it , not cast to double/int, in [visitArithmeticExpr] when getCommonType return Type.INVALID
> > > 
> > > 
> > > if common type is Type.INVALID, it's should check failed in line 626, any type can't cast to Type.INVALID, I think should to resolve this question...
> > 
> > 
> > @Seaven ok, i will modify. Also i think fn need to check failed when is null, maybe cause NPE
> 
> check function is the correct way, but function is null in here is an unexpected error, I suggest add `Preconditions.checkNotNull(fn)` directly
@Seaven if use Preconditions.checkNotNull(fn), also wiil throw Unknown error in mysql client

**Review by Seaven — DISMISSED** [review]

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -618,6 +618,11 @@ public Void visitArithmeticExpr(ArithmeticExpr node, Scope scope) {
                         throw unsupportedException("Unknown arithmetic operation " + op + " in: " + node);
                 }
 
+                if (Type.INVALID.equals(node.getChild(0).getTrueOriginType()) ||
+                        Type.INVALID.equals(node.getChild(1).getTrueOriginType())) {
+                    throw new SemanticException("Any function type can not cast to " + Type.INVALID.toSql());
+                }
```
[the remark this answers is not in the document]
Comment (leoyy0316) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java]: @Seaven yes, ut return error, modify it. done, thanks

**Review by Seaven — DISMISSED** [review]

**Comment by Seaven** [comment]
> @Seaven hi, when modify, some ut failed, can you help check this ut really need to check Type.INVALID to failed ? thanks

should move check from 621 to 629

**Comment by leoyy0316** [comment]
> > @Seaven hi, when modify, some ut failed, can you help check this ut really need to check Type.INVALID to failed ? thanks
> 
> should move check from line:621 to line:629

@Seaven  oh, i know, thanks

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|2|3|66.67%|[647]|

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**